### ISSUE TITLE
Installation issue on Mac OS

### ISSUE THREAD

**Issue by glagnese** [QuSpin/QuSpin#700]
Dear all,

I am trying to install the package on my Mac, but for reasons I do not understand I am failing both with the Conda installation and the manual one.

The command  "conda install -c weinbe58 quspin" prints
    PackagesNotFoundError: The following packages are not available from current channels:

      - quspin

    Current channels:

      - https://conda.anaconda.org/weinbe58
      - defaults
      - https://conda.anaconda.org/conda-forge

(My anaconda version is 24.7.1)

With the manual installation, through the command "python setup.py install --default-compiler-flags" I get the message
      os.environ["CFLAGS"]=os.environ["CFLAGS"]+" -O3 -march=native"
                            ~~~~~~~~~~^^^^^^^^^^
     File "<frozen os>", line 685, in __getitem__
   KeyError: 'CFLAGS'

The problem should be that the variable "CFLAGS" is not set. However, even if I were to set it manually, I am not sure how to do it properly.

Thanks for your help!

**Comment by KristianMunnikhuis** [comment]
This is also an issue I've been experiencing trying to run on MacOS as well as windows. I ended up getting it to work following #665 on Linux.

**Comment by KristianMunnikhuis** [comment]
> Thanks, I am going to try as you suggest

I'm not sure if these steps are entirely correct or necessary, but these are the exact steps I took to get Quspin working on a fresh install of Linux Mint 22. Previously, when I replied to you, I had been using a virtual machine which for some reason behaved differently than mine.



 1.clone the quspin repo and pull the most recent (currently dev_0.3.8) branch

 2. create a new conda environment using the packages in the attached txt file as follows:

`conda create -n <environment-name> --file quspin-linux_reqs.txt`

3. activate this environment

4.  run command 'python setup.py install build_ext -i --omp'

This command may be unnecessary and takes a long time to run. After doing 4 as #665 suggested, I would run into circular import errors. However, you should still check. Below are what I used to get it working after the errors.

5.  run (in the environment) 'conda install python=3.10'
6. manually copy the the quspin folder into miniconda3/envs/<environment name>/ lib/python3.10/site-packages/

After doing this, I was able to run 'import quspin' in python with no errors. 

Perhaps someone more experienced can give more details, hope this helps!

**Comment by glagnese** [comment]
Thanks four your help. 

So my first problem is that the command
conda create -n <environment-name> --file quspin-linux_reqs.txt

it's not working. Conda says that it does not find the packages in the file

Also, I have tried with 
python setup.py install --omp --default-compiler-flags --record install_file.txt
I don't get the CFLAGS error, but in the end I get 
Cython.Compiler.Errors.CompileError

**Comment by KristianMunnikhuis** [comment]
` conda create -n --file quspin-linux_reqs.txt`

Is this literally what you typed in? Did you name your environment? Did you cd to the directory holding the file?

**Comment by glagnese** [comment]
I gave a name to the environment, and I did cd. What I get is "PackagesNotFoundError: The following packages are not available from current channels:"  followed by all the packages in the "quspin-linux_reqs.txt" file

**Comment by mgbukov** [comment]
this should have been resolved with the `quspin 1.0.0` release; 

note the new `pip install quspin` installation procedure (conda is discontinued for the time being)

if fixed, please close the issue.
